src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,157 @@
+import React, {
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
+
+import { TodoList } from '../TodoList';
+import { apiClient } from '../../api/todos';
+import { useTodosContext } from '../store';
+import { ErrorOption } from '../../enum/ErrorOption';
+import { TodoFooter } from '../TodoFooter';
+import { TodoItem } from '../TodoItem';
+import { Todo } from '../../types/Todo';
+
+const USER_ID = 12027;
+
+export const TodoApp: React.FC = () => {
+  const {
+    todos,
+    addTodo,
+    hasError,
+    setError,
+    recieveTodos,
+    resetHasError,
+    toggleAllTodoCondition,
+  } = useTodosContext();
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [inputValue, setInputValue] = useState('');
+  const [isPostRequest, setIsPostRequest] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, []);
+
+  useEffect(() => {
+    resetHasError();
+
+    apiClient.getTodos(USER_ID)
+      .then(todosFS => {
+        recieveTodos(todosFS);
+      })
+      .catch(() => setError(ErrorOption.RecivingError))
+      .finally(() => setIsLoading(false));
+  }, [resetHasError, recieveTodos, setError]);
+
+  const onSubmitForm = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (inputValue.trim() === '') {
+      setError(ErrorOption.TitleError);
+
+      return;
+    }
+
+    setIsPostRequest(true);
+
+    setTempTodo({
+      id: 0,
+      userId: USER_ID,
+      title: inputValue,
+      completed: false,
+    });
+
+    const newTodo = {
+      userId: USER_ID,
+      title: inputValue,
+      completed: false,
+    };
+
+    apiClient.postTodo(newTodo)
+      .then(newTodoFS => {
```

Comment:
What is `FS`?
It is worth giving such a naming that does not raise questions, for this, try not to shorten the name to unreadable